### PR TITLE
[UX] Prevent Display from sleep when playing a game.

### PR DESCRIPTION
**Pull request by flavioislima**
Just add a new handler to prevent the display to go to sleep as asked on #1625 
Added a small fix for #1624  following what @CommandMC said on discord.
I could not reproduce the issue but changing the line to use `writeFileSync` didn't break the functionality.

---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [x] Tested the feature and it's working on a current and clean install.
- [x] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

### CODE REVIEW

**Comment by Etaash-mathamsetty** [comment]
It doesn't deactivate when I stop playing the game, but steam doesn't even care if the game is running or not so I think it's fine for now

**Comment by flavioislima** [comment]
> It doesn't deactivate when I stop playing the game, but steam doesn't even care if the game is running or not so I think it's fine for now

That's weird because Heroic removes the lock and stop the powersaver blocker so it should get back to normal behavior.
Not a major issue but should work.

**Comment by flavioislima** [comment]
@Etaash-mathamsetty 
pushed a fix, the method to stop the blocker was not always being called, now its working as it should.

**Comment by Etaash-mathamsetty** [comment]
> @Etaash-mathamsetty pushed a fix, the method to stop the blocker was not always being called, now its working as it should.

It's still not working for me...

**Comment by flavioislima** [comment]
> > @Etaash-mathamsetty pushed a fix, the method to stop the blocker was not always being called, now its working as it should.
> 
> It's still not working for me on kde...

Can you run it from terminal and check if it shows a message like: _stopping display power saver blocker_ or something?

**Comment by Etaash-mathamsetty** [comment]
all I get is this
`(16:41:00) INFO:    [Backend]:         Preventing display from sleep`
I do get an error from discord rich presence since I don't have discord installed anymore (I am using webcord), so let me try disabling the discord integration thing

**Comment by flavioislima** [comment]
@Etaash-mathamsetty I totally forgot that KDE showed this information.
I was actually trapped inside a Javascript issue.
The `powerDisplayId` was coming as `0` and since I was checking `if(powerDisplayId)` JS was returning false and not stopping the blocker, changing it to check if it was null fixed it. Now I can see that it disappears from the blocker list, but try again on your machine.

**Comment by Etaash-mathamsetty** [comment]
> @Etaash-mathamsetty I totally forgot that KDE showed this information. I was actually trapped inside a Javascript issue. The `powerDisplayId` was coming as `0` and since I was checking `if(powerDisplayId)` JS was returning false and not stopping the blocker, changing it to check if it was null fixed it. Now I can see that it disappears from the blocker list, but try again on your machine.

works fine now

File: src/components/UI/SearchBar/index.tsx [hunk 1]
```diff
@@ -11,6 +11,11 @@ import ContextProvider from 'src/state/ContextProvider'
 import FormControl from '../FormControl'
 import './index.css'
 
+function fixFilter(text: string) {
+  const regex = new RegExp(/([\\|*|+|(|)|[|]|])+/, 'g')
```
Comment (arielj) [inline src/components/UI/SearchBar/index.tsx:15]: we should consider `?` too in this regexp

Comment (flavioislima) [reply src/components/UI/SearchBar/index.tsx:15]: ok will add it

**Review by arielj — APPROVED** [review]
looks good, I haven't had issues with powersaver / screensaver while playing so I don't know how to test this

I was a bit confused about the translations and the extra changes like the searchbar fixes, I think we should try to keep separated fixes in separated PRs so things are easier to follow (I don't think we need to split this PR, but would be nice to not have a mix of unrelated changes xD)

**Pull request merged**